runtime/parachains/src/paras_inherent.rs

```diff
@@ -868,169 +867,243 @@ fn sanitize_backed_candidates<
 
 		#[test]
-		fn all_backed_candidates_get_truncated() {
+		fn include_backed_candidates() {
 			new_test_ext(MockGenesisConfig::default()).execute_with(|| {
-				let backed_candidates = vec![BackedCandidate::default(); 10];
-				let max_block_weight =
-					<Test as frame_system::Config>::BlockWeights::get().max_block;
-				// if the consumed resources are precisely equal to the max block weight, we do not truncate.
-				System::set_block_consumed_resources(max_block_weight + 1, 0);
-				assert_eq!(limit_backed_candidates::<Test>(backed_candidates).len(), 0);
+				let dispute_statements = BTreeMap::new();
+
+				let mut backed_and_concluding = BTreeMap::new();
+				backed_and_concluding.insert(0, 1);
+				backed_and_concluding.insert(1, 1);
+
+				let scenario = make_inherent_data(
+					TestConfig {
+						dispute_statements,
+						dispute_sessions: vec![0, 0,],
+						backed_and_concluding,
+						num_validators_per_core: 1,
+						includes_code_upgrade: false
+					}
+				);
+
+				// We expect the scenario to have cores 0 & 1 with pending availability. The backed
+				// candidates are also created for cores 0 & 1, so once the pending available
+				// become fully available those cores are marked as free and scheduled for the backed
+				// candidates.
+				let expected_para_inherent_data = scenario.data.clone();
+
+				// Check the para inherent data is as expected:
+				// * 1 bitfield per validator
+				assert_eq!(expected_para_inherent_data.bitfields.len(), 2);
+				// * 1 backed candidate per core
+				assert_eq!(expected_para_inherent_data.backed_candidates.len(), 2);
+				// * 0 disputes.
+				assert_eq!(expected_para_inherent_data.disputes.len(), 0);
+				let mut inherent_data = InherentData::new();
+				inherent_data
+					.put_data(PARACHAINS_INHERENT_IDENTIFIER, &expected_para_inherent_data)
+					.unwrap();
+
+				// The current schedule is empty prior to calling `create_inherent_enter`.
+				assert_eq!(<scheduler::Pallet<Test>>::scheduled(), vec![]);
+
+				// Nothing is filtered out (including the backed candidates.)
+				assert_eq!(
+					Pallet::<Test>::create_inherent_inner(&inherent_data.clone()).unwrap(),
+					expected_para_inherent_data
+				);
+
+				// The schedule is still empty prior to calling `enter`. (`create_inherent_inner` should not
+				// alter storage, but just double checking for sanity).
+				assert_eq!(<scheduler::Pallet<Test>>::scheduled(), vec![]);
+
+				assert_eq!(Pallet::<Test>::on_chain_votes(), None);
+				// Call enter with our 2 backed candidates
+				assert_ok!(Pallet::<Test>::enter(
+					frame_system::RawOrigin::None.into(),
+					expected_para_inherent_data
+				));
+				assert_eq!(
+					// The length of this vec is equal to the number of candidates, so we know our 2
+					// backed candidates did not get filtered out
+					Pallet::<Test>::on_chain_votes().unwrap().backing_validators_per_candidate.len(),
+					2
+				);
 			});
 		}
 
 		#[test]
-		fn ignores_subsequent_code_upgrades() {
```

Comment:
I am a little confused by this diff, did this test get deleted?

Reply:
Yes, these tests are no longer valid and they did not test the proper things to begin with.